GTHC/web
# Calendar and Shift Model updates for basic Calendar functionality

### Purpose
The overall purpose of this PR is to build the main functionality of the calendar/scheduler. This includes everything both frontend and backend.

### Tasks
- [x] Create Model and Controller for Shifts
   - this also involves setting up DB relations between the Shift model and other models such as User and Team
- [x] Create all endpoints for Shift API
    - this includes all GET, PUT/PATCH, POST, and DELETE requests that involve receiving shift data and any other data necessary for the calendar
- [x] Fix Big Calendar bug
    - finally fixed long time bug that involved multiple copies of React being loaded onto app and blocking Big Calendar component
- [x] Show Shifts on Calendar
- [x] Implement all API endpoints onto frontend
    - this also includes having the user interface to interact with the API properly (having a button and modal to create new shifts, or button and text fields to update shifts etc.)
    - all endpoints include at least: GET all shifts, POST new shift, PUT/PATCH update shift, and POST new user and shift relation (add a user to a shift).

### Test
How to test:
1. Start application
   - if it is your first time running:
        - run `yarn dev:setup`
   - if you have run app before:
        - run `yarn run flush`
        - respond to first prompt with `y`
        - run `yarn dev:start`
2. Sign up to an existing team (since fake shift data has been already been generated, but there is no problem with signing up and creating a team).
3. Go to Calendar tab
4. Click on "shifts" to see expanded shift data (NOT final design, just to show data can be displayed)
5. Click on "Update" button to see update modal (NOT final design nor is the API endpoint implemented so changes are not saved, see comment below regarding this)
6. Drag mouse on calendar to create a new shift.
7. Enter data for new shift (add a description, title, and add users), and save (see new shift appear on the calendar!).
